### PR TITLE
Enforce react-hooks/exhaustive-deps lint rule

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

* Instead of showing a warning when `react-hooks/exhaustive-deps` lint rule is not followed - throw error instead
* Fix `react-hooks/exhaustive-deps` issues in payments components

### CODE REVIEW

**Comment by MananTank** [comment]
@IDubuque Can you verify that the useEffect deps changes I have made in this PR do not add unwanted side-effects?

**Comment by jnsdls** [comment]
waiting on @IDubuque review

File: packages/react/src/payments/components/common/Modal.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const Modal: React.FC<{
     return () => {
       document.removeEventListener("keydown", keyDownHandler);
     };
-  }, []);
+  }, [escapeToClose, onClose]);
```
Comment (jnsdls) [inline packages/react/src/payments/components/common/Modal.tsx:39]: this one is fine

File: packages/react/src/payments/components/VerifyOwnershipWithPaper.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ export const VerifyOwnershipWithPaper: React.FC<
     return () => {
       window.removeEventListener("message", handleMessage);
     };
-  }, []);
+  }, [onError, onSuccess, onWindowClose]);
```
Comment (jnsdls) [inline packages/react/src/payments/components/VerifyOwnershipWithPaper.tsx:87]: this one is fine

File: packages/react/src/payments/components/PaperCheckout.tsx [hunk 3]
```diff
@@ -141,7 +141,7 @@ export const PaperCheckout = <T extends ContractType>({
     return () => {
       window.removeEventListener("message", handleMessage);
     };
-  }, []);
+  }, [onCloseCheckout, onPaymentSuccess, onTransferSuccess]);
```
Comment (jnsdls) [inline packages/react/src/payments/components/PaperCheckout.tsx:144]: this one is fine

File: packages/react/src/payments/components/CheckoutWithCard.tsx [hunk 4]
```diff
@@ -97,7 +97,20 @@ export const CheckoutWithCard = ({
...
+    configs,
+    locale,
+    onBeforeModalOpen,
+    onCardDetailLoad,
+    onError,
+    onPaymentSuccess,
+    onPriceUpdate,
+    onReview,
+    options,
+    sdkClientSecret,
```
Comment (jnsdls) [inline packages/react/src/payments/components/CheckoutWithCard.tsx:112]: this one needs a deeper look

**Pull request merged**